frontend/src/modules/entitydetails/components/Metadata.js

```diff
@@ -25,19 +25,58 @@ type Props = {|
     +navigateToPath: (string) => void,
 |};
 
+type State = {|
+    seeMore: boolean,
+    isNotTruncated: boolean
+|};
+
 
 /**
  * Component showing metadata of an entity.
  *
  * Shows:
  *  - the original string
  *  - a comment (if any)
+ *  - a group comment (if any)
+ *  - a resource comment (if any)
  *  - a context (if any)
  *  - a list of source files (if any)
  *  - a link to the resource
  *  - a link to the project
  */
-export default class Metadata extends React.Component<Props> {
+export default class Metadata extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            seeMore: false,
+            isNotTruncated: false
+        };
+        this.span = React.createRef();
+    }
+
+    componentDidMount() {
+        this.setState({
+            isNotTruncated: this.isEllispsisActivated()
+            });
+        }
```

Comment:
This method has weird alignment. See `componentDidUpdate` for reference (including `setState`).